test/validation/api/classification/odp_classification_test_pmr.c

```diff
@@ -2293,6 +2317,14 @@ static void test_pmr_series(const int num_udp)
 	CU_ASSERT_FATAL(pkt != ODP_PACKET_INVALID);
 	CU_ASSERT(seqno == cls_pkt_get_seq(pkt));
 	CU_ASSERT(retqueue == queue_ip);
+
+	if (marking) {
```

Comment:
I am just thinking that should there be a test case also for the case where first level PMR matches but the second level PMRs do not. All PMRs would be configured to do the marking and the expected mark value would be the one of the firt level PMR. The current test would not notice if the implementation wrote the mark whenever it compared the packet with a PMR instead of only setting the mark upon PMR match.

Reply:
I think this could be another test case for parallel/serial PMR usage in general, and include marking test cases into it. However, I'll add cls_mark default value tests to packet validation suite.